File: src/main/org/nlogo/prim/etc/_followme.java
package org.nlogo.prim.etc ;

import org.nlogo.agent.Turtle;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Syntax;
import org.nlogo.api.Perspective;

public final strictfp class _followme
	extends Command
{
	@Override
	public Syntax syntax()
	{
		return Syntax.commandSyntax( "-T--" , true ) ;
	}
	@Override
	public void perform( final Context context )
	{
		Turtle turtle = (Turtle) context.agent ;
		world.observer().setPerspective( Perspective.FOLLOW , turtle ) ;
		// the following code is duplicated in _follow and _followme - ST 6/28/05
		int distance = (int) turtle.size() * 5 ;
		world.observer().followDistance
			( StrictMath.max( 1 , StrictMath.min( distance , 100 ) ) ) ;
		context.ip = next ;
	}
}

File: src/main/org/nlogo/prim/etc/_followme.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.agent.Turtle
import org.nlogo.nvm.{ Command, Context }
import org.nlogo.api.{ Perspective, Syntax }

class _followme extends Command {
  override def syntax =
    Syntax.commandSyntax("-T--", true)
  override def perform(context: Context) {
    val turtle = context.agent.asInstanceOf[Turtle]
    world.observer.setPerspective(Perspective.Follow, turtle)
    context.ip = next
  }
}
